src/tests/infraestructure/controller/TestController.ts

```diff
@@ -0,0 +1,193 @@
+      const status = errorMessage === 'Test not found' ? 404 : 500;
+      res.status(status).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async startTest(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+      const { userId } = req.body;
+
+      if (!id) {
+        res.status(400).json({
+          success: false,
+          error: 'Test ID is required',
+        });
+        return;
+      }
+
+      if (!userId) {
+        res.status(400).json({
+          success: false,
+          error: 'userId is required',
+        });
+        return;
+      }
+
+      const result = await services.tests.start.execute(id, userId);
+      res.status(201).json({
+        success: true,
+        data: result,
+      });
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+      const status = errorMessage.includes('not found')
+        ? 404
+        : errorMessage.includes('not active')
+          ? 400
+          : 500;
+      res.status(status).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async getQuestions(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+      const { sessionId } = req.query;
+
+      if (!id) {
+        res.status(400).json({
+          success: false,
+          error: 'Test ID is required',
+        });
+        return;
+      }
+
+      if (!sessionId || typeof sessionId !== 'string') {
+        res.status(400).json({
+          success: false,
+          error: 'sessionId is required',
+        });
+        return;
+      }
+
+      const result = await services.tests.getQuestions.execute(id, sessionId);
+      res.status(200).json({
+        success: true,
+        data: result,
+      });
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+      const status = errorMessage.includes('session') ? 400 : 500;
+      res.status(status).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async submitTest(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+      const { sessionId, answers } = req.body;
+
+      if (!id) {
+        res.status(400).json({
+          success: false,
+          error: 'Test ID is required',
+        });
+        return;
+      }
+
+      if (!sessionId) {
+        res.status(400).json({
+          success: false,
+          error: 'sessionId is required',
+        });
+        return;
+      }
+
+      if (!answers || !Array.isArray(answers)) {
+        res.status(400).json({
+          success: false,
+          error: 'answers array is required',
+        });
+        return;
+      }
+
+      const result = await services.tests.submit.execute(
```

Comment:
submitTest calls services.tests.submit.execute(id, sessionId, answers) without validating that the authenticated user owns the session or that the session hasn't expired (expiry check is missing in use-case as well).

**Sugerencia:**
```typescript
Require authentication, pass req.user.id to the use-case (or validate session.userId === req.user.id) and ensure expiry checks are in place. Map submission-specific domain errors to appropriate HTTP statuses (409 for duplicate, 400 for invalid data). Priority: critical (security / correctness)
```